pkg/controller/petset/pet.go

```diff
@@ -251,8 +251,8 @@ func (p *apiServerPetClient) SyncPVCs(pet *pcb) error {
 		if !exists {
 			if err := p.createPVC(&pet.pvcs[i]); err != nil {
 				errMsg += fmt.Sprintf("Failed to create %v: %v", pvc.Name, err)
+				p.event(pet.parent, "Create", fmt.Sprintf("pvc: %v", pvc.Name), err)
 			}
-			p.event(pet.parent, "Create", fmt.Sprintf("pvc: %v", pvc.Name), err)
```

Comment:
I see. I was under the impression that only failures would create an event in that case. Okay, that makes sense. My feeling was that we could keep `getPVC(...)` simple and have it always return the error and no additional boolean for whether it exists. We classify that error using `errors.IsNotFound(...)` later when we need to make a decision about it within `SyncPVCs(...)`. Does that make sense?